teachertool/src/components/AddressBar.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+// eslint-disable-next-line import/no-internal-modules
+import css from "./styling/AddressBar.module.scss";
+
+import { useContext, useState, useMemo, useCallback, useEffect } from "react";
+import { AppStateContext } from "../state/appStateContext";
+import { classList } from "react-common/components/util";
+import { Input } from "react-common/components/controls/Input";
+import { loadProjectMetadataAsync } from "../transforms/loadProjectMetadataAsync";
+
+interface IProps {
+    className?: string;
+}
+
+export const AddressBar: React.FC<IProps> = ({ className }) => {
```

Comment:
I chose AddressBar because:
1. it's modeled closely after a browser's address bar.
2. the most common input will be a pasted URL, like an address bar.

I am open to alternatives!

Reply:
ShareLinkInput would be the most descriptive, probably. But I'm fine with AddressBar.

Reply:
Renamed!